tests/src/test/scala/org/http4s/ResponderSpec.scala

```diff
@@ -20,102 +20,105 @@ import cats.effect.IO
-      wHeader.headers.get(Connection) must beSome(Connection("close".ci))
-
-      val newHeaders = wHeader.removeHeader(Connection)
-      newHeaders.headers.get(Connection) must beNone
-    }
-
-    "Replace all headers" in {
-      val wHeader =
-        resp.putHeaders(Connection("close".ci), `Content-Length`.unsafeFromLong(10), Host("foo"))
-      (wHeader.headers.toList must have).length(3)
-
-      val newHeaders = wHeader.withHeaders(Date(HttpDate.Epoch))
-      (newHeaders.headers.toList must have).length(1)
-      newHeaders.headers.get(Connection) must beNone
-    }
-
-    "Replace all headers II" in {
-      val wHeader =
-        resp.putHeaders(Connection("close".ci), `Content-Length`.unsafeFromLong(10), Host("foo"))
-      (wHeader.headers.toList must have).length(3)
-
-      val newHeaders = wHeader.withHeaders(Headers.of(Date(HttpDate.Epoch)))
-      (newHeaders.headers.toList must have).length(1)
-      newHeaders.headers.get(Connection) must beNone
-    }
-
-    "Filter headers" in {
-      val wHeader =
-        resp.putHeaders(Connection("close".ci), `Content-Length`.unsafeFromLong(10), Host("foo"))
-      (wHeader.headers.toList must have).length(3)
-
-      val newHeaders = wHeader.filterHeaders(_.name != "Connection".ci)
-      (newHeaders.headers.toList must have).length(2)
-      newHeaders.headers.get(Connection) must beNone
-    }
-
-    "Set cookie from tuple" in {
-      resp.addCookie("foo", "bar").cookies must_== List(ResponseCookie("foo", "bar"))
-    }
-
-    "Set cookie from Cookie" in {
-      resp.addCookie(ResponseCookie("foo", "bar")).cookies must_== List(
-        ResponseCookie("foo", "bar"))
-    }
-
-    "Set multiple cookies" in {
+  test("Responder should Change status") {
+    val resp = Response[IO](Status.Ok)
+
+    assertEquals(resp.status, Status.Ok)
+
+    assertEquals(resp.withStatus(Status.BadGateway).status, Status.BadGateway)
+  }
+
+  test("Responder should Replace content type") {
+    assert(resp.contentType.isEmpty)
+    val c1 = resp
+      .putHeaders(`Content-Length`.unsafeFromLong(4))
+      .withContentType(`Content-Type`(MediaType.text.plain))
+      .putHeaders(Host("foo"))
+
+    assertEquals(c1.headers.count(_.is(`Content-Type`)), 1)
+    assertEquals(c1.headers.count(_.is(`Content-Length`)), 1)
+    assertEquals(c1.headers.toList.length, 3)
+    assertEquals(c1.contentType, Some(`Content-Type`(MediaType.text.plain)))
+
+    val c2 = c1.withContentType(`Content-Type`(MediaType.application.json, `UTF-8`))
+
+    assertEquals(c2.contentType, Some(`Content-Type`(MediaType.application.json, `UTF-8`)))
+    assertEquals(c2.headers.count(_.is(`Content-Type`)), 1)
+    assertEquals(c2.headers.count(_.is(`Content-Length`)), 1)
+    assertEquals(c2.headers.count(_.is(Host)), 1)
+  }
+
+  test("Responder should Remove headers") {
+    val wHeader = resp.putHeaders(Connection("close".ci))
+    assertEquals(wHeader.headers.get(Connection), Some(Connection("close".ci)))
```

Comment:
might cause problems on Scala 3, needs a val to be able to type it correctly in the HeaderKey#HeaderT

Reply:
thanks, https://github.com/http4s/http4s/pull/4437/commits/e20e70e0feb09a1675ea3ef2d17edcb14990b00e